properties/test_encode_decode.py

```diff
@@ -43,3 +45,10 @@ def test_CFScaleOffset_coder_roundtrip(original) -> None:
     coder = xr.coding.variables.CFScaleOffsetCoder()
     roundtripped = coder.decode(coder.encode(original))
     xr.testing.assert_identical(original, roundtripped)
+
+
+# TODO: add cftime.datetime
+@given(dt=st.datetimes())
+def test_iso8601_decode(dt):
```

Comment:
hehe datetimes are limited between year=1 and year=9999 so it doesn't change your additional changes here, but good to add anyway I guess.